Pass variables to replace tokens in query

rafatower opened the pull request:
This PR fixes #679. It comprises the following changes:
- the patch in https://github.com/CartoDB/node-mapnik/pull/5/commits/5182b7168f3c460cf19b94d98e9cfb9f9a4512f2 cherry-picked on master
- an end-to-end test for it
- the dependency `mapnik-vector-tile` updated to current master commit (to be changed).

Some remarks:
- As the change is built on https://github.com/mapbox/mapnik-vector-tile/pull/248 and takes advantage of the optional second argument of the vector tile processor constructor, updating to a newer version of `mapnik-vector-tile` is required, otherwise it will fail at build time. @springmeyer do you plan to release a `1.4.1` or `1.5.0` version anytime soon?
- The test depends on having postgres+postgis installed. Otherwise it is skipped. I'd like to add it to CI builds, if possible.

Comments & suggestions welcomed.

springmeyer commented:
Noticed this is failing currently due to unrelated regression to ndoe v0.10.x support. Please rebase from master once https://github.com/mapnik/node-mapnik/pull/810 is merged.

> @springmeyer do you plan to release a 1.4.1 or 1.5.0 version anytime soon?

mapnik-vector-tile@1.5.0 is now published.

> The test depends on having postgres+postgis installed. Otherwise it is skipped. I'd like to add it to CI builds, if possible.

👍  And ideally on OSX as well as Travis. As you know postgres/postgis is a challenging install. I would prefer not to depend on third-party launchpad PPAs that change over time and may break unpredictably. For this reason I would prefer postgres/postgis is installed from mason, which I can update if needed to keep node-mapnik tests passing against different versions. Want to have a look at https://github.com/springmeyer/postgis-mason-install for an example of how to install this on travis? Or provide an alternative strategy?

rafatower commented:
I rebased from master, updated `mapnik-vector-tile@1.5.0` (thanks for the prompt release BTW), added the scripts to install postgis from mason and integrated them with travis build. Hope this is now ready to go. Thanks!

springmeyer commented on test/vector-tile.postgis.test.js:
```diff
@@ -0,0 +1,87 @@
+
+    it('passes variables to replace tokens in query', function(done) {
+        if (!hasPostgisAvailable) {
+            this.skip('postgis not availabel');
```
Minor spelling fix needed: `availabel` -> `available`.

springmeyer commented:
@rafatower thanks for putting together the testing quickly. Looks great. I am headed on vacation for a week, but I will plan to test this myself and merge when back.

springmeyer commented:
> I will plan to test this myself and merge when back.

Sorry I've not gotten to this. I've become too busy with other responsibitilies to dive into this code with the attention I feel is needed. This is a very cool feature, but needs testing to ensure it does not break anything in an unintended way. Is there anyone else reading this could provide:

 - a second set of eyes?
 - test the feature in their downstream application to ensure it works?
 - help ensure all the tests are passing on travis with this PR?

springmeyer commented:
Going to merge this into master without testing further. This is a rare exception for me, but the change is primarily tests so this should be safe. Please ping here if any hits any problems due to this change and we'll investigate.
